### ISSUE TITLE
scrollTo issues inside an iframe

### ISSUE THREAD

**Issue by jennifer-shehane** [flesler/jquery.scrollTo#152]
Hi, I am having a couple of issues with `scrollTo` inside of an iframe. There are a few reasons behind this for me. 

---

1. I am passing the iframe's `window` as the `element` argument to `scrollTo`. When running the [WinMaxY-to-iframe-compat.html](https://github.com/flesler/jquery.scrollTo/blob/master/tests/WinMaxY-to-iframe-compat.html) with this updated use case, the scrollTo does not work. 

**To reproduce in tests:**

*test.js*
```js
// does not scroll
var iframeWin = this[0].contentWindow

return $(iframeWin).scrollTo('max', 1000).scrollTo(0, 1000);
```

The issue is that within `scrollTo.js`, [L69](https://github.com/flesler/jquery.scrollTo/blob/master/jquery.scrollTo.js#L69), the code is looking at `this.contentWindow`(which returns undefined -- we passed in the iframe's window, which obv has no contentWindow), so the code then sets `elem` to `window`. Unfortunately this `window` is the outer `window`, so the context is incorrect in the code from then on.

This is easily enough solved by setting my iframe window's `contentWindow` to it's own iframe `window` in my code.

*my updated test.js*
```js
// does scroll
var iframeWin = this[0].contentWindow

iframeWin.contentWindow = iframeWin

return $(iframeWin).scrollTo('max', 1000).scrollTo(0, 1000);
```

---

2. Regardless of the issue above, `scrollTo` breaks within an iframe when the outer window's jQuery version is 2.0.0 or above. The first axis' scrolling will always be ignored. 

**To reproduce in tests:**

*WinMaxY-to-iframe-compat.html*
```html
<!-- we updated our jquery to 2.0.0 -->
<script type="text/javascript" src="https://code.jquery.com/jquery-2.0.0.js"></script>
```

*test.js*
```js
// the example below scrolls 550px vertically (on y axis) ONLY
// fails to scroll on x axis
this.scrollTo({ left: 550, top: 550 }, { axis: 'xy' });

// the example below scrolls 550px horizontally (on x axis) ONLY
// fails to scroll on y axis
this.scrollTo({ left: 550, top: 550 }, { axis: 'yx' });
```

**Comment by jennifer-mann** [comment]
I actually found the culprit of the 2nd issue. 

In jQuery 2.0.0+ to 2.2.0, when `$(window).scrollTop()` or $(window).scrollLeft()` are called (as they are within `scrollTo` [L202](https://github.com/flesler/jquery.scrollTo/blob/master/jquery.scrollTo.js#L202)), jQuery was incorrectly calculating the values sent to `window.scrollTo()`. The other axis was being incorrectly calculated off of the parent window's `pageXOffset` or `pageYOffset` and not the iframe's window. This resulted in `scrollTo`'s `scrollTop` and `scrollLeft` calls always setting the other axis to `0` for every call.

**This means that you can never scroll both axes within an iframe using the `scrollTo` library with jQuery versions 2.0.0-2.1.4.**

The issue was fixed in jQuery version 2.2.0 however. The fixed lines of code are referenced [here](https://github.com/jquery/jquery/commit/d21edb599d8f5f80a3f3ecba5b62311b9cd1a3e6) in jQuery's library.

**Comment by flesler** [comment]
Thanks for the find and glad it's not a scrollTo issue
